### PR TITLE
pod-scaler: add e2e test for build pod admission

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This sets the stage for an e2e for pod resource mutation.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Part of Part of DPTP-2349

File: cmd/pod-scaler/main.go [hunk 1]
```diff
@@ -212,9 +214,20 @@ func mainProduce(opts *options, cache cache) {
 }
 
 func mainAdmission(opts *options) {
-	restConfig, err := rest.InClusterConfig()
+	// TODO: figure out the logrusr dependency hell with logr and c-r
```
Comment (stevekuznetsov) [inline cmd/pod-scaler/main.go:217]: @alvaroaleman do you have any suggestions here? Importing `logrusr` (a shim to allow `logrus` to be the `logr` logger used for `controller-runtime`) pulls in an incompatible version of `logr` and causes the `controller-runtime` vendors to not compile :|

Comment (alvaroaleman) [reply cmd/pod-scaler/main.go:217]: Copy the shim or `replace` the `logr` version

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**